src/storage.cc

```diff
@@ -209,9 +210,16 @@ Status Storage::CreateColumnFamilies(const rocksdb::Options &options) {
     tmp_db->Close();
     delete tmp_db;
   }
-  // Open db would be failed if the column families have already exists,
```

Comment:
Is it possible we check exact that it failed because of "already exists" instead of "NOT column families not opened"? I'm afraid there is still some edge case we miss.

Also, shall we prefer C++ idiom and thus check strings with `<string>`? That means:

* `s.ToString().rfind(notOpenedPrefix, 0) != 0`

over

* `!strncasecmp(s.ToString().c_str(), notOpenedPrefix.c_str(), notOpenedPrefix.size())`

Reply:
> Is it possible we check exact that it failed because of "already exists" instead of "NOT column families not opened"? I'm afraid there is still some edge case we miss.

Yes, it's a bit tricky and confuse, but we can't change the status message AFAIK since it returned by the RocksDB.

For the second suggestion, We do prefer C++ idiom. my initial intention was to ignore the case when comparing the prefix. But seems this's unnecessary since it can't protect anything if the status message is modified, I will change to `rfind` to simplify the expression.

Reply:
Sorry I just (again, haha) revert the boolean logic :P

Ping me when you have a conclusion on whether use rfind or keep strncasecmp.

Reply:
Thanks @tisonkun, changed to the c++ api and do a bit logic refactor, please take a look again.